src/python/dbs/apis/dbsClient.py

```diff
@@ -400,7 +401,16 @@ def __callServer(self, method="", params={}, data={}, callmethod='GET', content=
                 print("HTTP={} URL={} method={} params={} data={} headers={}".format(callmethod, self.url, method, params, data, request_headers))
             self.http_response = method_func(self.url, method, params, data, request_headers)
         except HTTPError as http_error:
-            self.__parseForException(http_error)
+            if http_error.code == 200:
```

Comment:
there are json format error returned besides http error code 200. See 
[#43](https://github.com/dmwm/DBSClient/pull/43).  It a more general error handling using the error body format than the error code.

Reply:
Yuyi, the point is that `__parseForException` function does not check if given `http_error` has JSON and what it does check is presence of `str(data).find("<html>")!=-1` but the HTTP response can be in different forms, it may not contain such pattern. Moreover, it can be proxy dependent, e.g. apache frontend can return one pattern while another frontend may return another. You either need to rely on HTTP codes and check for `Content-Type` (this is what I should add to PR) or leave response as is.

I'll change the PR to check for content-type instead of code then.

Reply:
@yuyiguo I changed code to check for content type. This is the most robust approach. The HTTP response from any server should provide content type. If it is JSON you can safely parse it, otherwise it is better to print it as is.